servlet-feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/bridge/servlet-api-bridge/main/module.xml

```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~ Copyright 2010, Red Hat, Inc., and individual contributors
+  ~ as indicated by the @author tags. See the copyright.txt file in the
+  ~ distribution for a full listing of individual contributors.
+  ~
+  ~ This is free software; you can redistribute it and/or modify it
+  ~ under the terms of the GNU Lesser General Public License as
+  ~ published by the Free Software Foundation; either version 2.1 of
+  ~ the License, or (at your option) any later version.
+  ~
+  ~ This software is distributed in the hope that it will be useful,
+  ~ but WITHOUT ANY WARRANTY; without even the implied warranty of
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  ~ Lesser General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU Lesser General Public
+  ~ License along with this software; if not, write to the Free
+  ~ Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+  ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+  -->
+
+<module xmlns="urn:jboss:module:1.7" name="org.wildfly.bridge.servlet-api-bridge">
+    <resources>
+        <artifact name="${org.wildfly.bridge:servlet-api-bridge}">
+            <conditions>
+                <property-not-equal name="ee8.preview.mode" value="true" />
```

Comment:
Ok, so this module is loaded when ee8.preview.mode is NOT enabled. The reason is to align with servlet 4.0 API?

Reply:
@jstourac I'll hazard a guess after glancing quickly. By default the implementation supports servlet 4.0, and the bridge is to support servlet 3.1 (or whatever it is in EE7). So the bridge gets loaded only when ee8 is not enabled as per option 2 of 'General Approach' in Stuart's document.

Reply:
Yes, this module is only loaded in EE7 mode. The reason is that the EE8 API introduces some new classes, and they need to be present for Undertow's Servlet implementation to work correctly.

Reply:
Thank you both for info!